sftp.c

```diff
@@ -934,7 +971,14 @@ do_globbed_ls(struct sftp_conn *conn, const char *path,
 			mprintf("%s\n", lname);
 			free(lname);
 		} else {
-			mprintf("%-*s", colspace, fname);
+			/* *** ICL Modification *** if file has .iron extension, prefix with lock icon */
+			int offset = iron_extension_offset(fname);
+			if (offset > 0) {
+				mprintf("%s%-*s", ICL_LOCK_ICON, (colspace - 2), fname);
+			} else {
+				mprintf("  %-*s", (colspace - 2), fname);
+			}
+			/* *** */
 			if (c >= columns) {
```

Comment:
It is in woefully many places, because there are woefully many places that print out the same thing. One is in the progress meter, one is in sftp-common.c:ls_file, and three are in sftp.c. Two are in do_ls_dir, where in two different code paths it generates the listing, but one is aligned into columns and the other is a long listing format, and the third is in do_globbed_ls. It looks like I could at least put that into a common routine - got sloppy.